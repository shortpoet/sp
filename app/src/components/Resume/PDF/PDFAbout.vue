<template>
<section class="resume-section" id="about">
<!-- basic info row -->
<div
class="d-flex flex-row justify-content-between align-items-center mt-1 ml-3"
id="basic-info"
>
<!-- photo column -->
<div class="d-flex flex-column align-items-center" id="photo">
<img
class="img-fluid img-profile rounded m-1 mb-2"
:src="pic"
alt="profile pic"
:style="{ height: '8rem', width: '7rem' }"
/>
</div>

<!-- name contact column -->
<div class="d-flex flex-column mt-1 mt-md-2 ml-2" id="name-contact">
<ul class="list-inline">
<li>
<h2 style="font-family: 'Saira Extra Condensed'" class="heading heading-2 mb-0">
{{ name }}
<span
id="pdf-surname"
style="font-family: 'Saira Extra Condensed'"
class="text-primary"
>{{ surname }}</span
>
</h2>
</li>
<li>
<span class="not-name">
<a style="font-family: 'Open Sans'" :href="'mailto:' + email">
{{ email }}
</a>
</span>
</li>
<li>
<span
id="pdf-address"
style="font-family: 'Saira Extra Condensed'"
class="not-name"
>
{{ address }}
</span>
</li>
<li>
<span id="pdf-flags" class="not-name">{{ flags.replace(/,/g, "") }}</span>
</li>
<li>
<span
id="pdf-visas"
style="font-family: 'Saira Extra Condensed'"
class="not-name"
>Valid Work Visas: {{ visas }}</span
>
</li>
</ul>
</div>

<!-- socials column -->
<div class="d-flex flex-column ml-auto mr-1 mr-md-3" id="socials">
<PDFSocials :renderPDF="renderPDF" :socials="socials" />
</div>
</div>
</section>
</template>

<script>
// import PDFAction from '@/components/Utils/PDFAction'
import PDFSocials from "@/components/Resume/PDF/PDFSocials";

export default {
name: "PDFAbout",
components: {
// PDFAction,
PDFSocials,
},
props: {
name: {
type: String,
},
surname: {
type: String,
},
email: {
type: String,
},
address: {
type: String,
},
visas: {
type: String,
},
flags: {
type: String,
},
renderPDF: {
type: Boolean,
default: false,
},
socials: {
type: Array,
},
},
data() {
return {
pic: require("@/assets/github_profile_pic.jpg"),
};
},
computed: {},
mounted() {},
};
</script>
<style>
.social-item {
font-size: 0.55rem;
}
</style>
