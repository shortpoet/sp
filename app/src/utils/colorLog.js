const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
