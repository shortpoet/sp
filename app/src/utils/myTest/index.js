import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
