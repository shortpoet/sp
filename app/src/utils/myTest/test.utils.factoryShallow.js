import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow