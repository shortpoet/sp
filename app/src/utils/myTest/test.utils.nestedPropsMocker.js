

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
