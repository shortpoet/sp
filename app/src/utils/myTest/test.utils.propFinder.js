import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder