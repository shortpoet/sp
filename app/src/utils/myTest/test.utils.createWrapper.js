import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper