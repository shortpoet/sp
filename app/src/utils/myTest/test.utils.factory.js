import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory