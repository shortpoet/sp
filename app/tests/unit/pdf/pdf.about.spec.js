import * as myTest from '@/utils/myTest'
const { textMatcherFactory, createWrapper, propsMocker, propMocker } = myTest.default

import PDFAbout from '@/components/Resume/PDF/PDFAbout'
import { cloneDeep } from 'lodash'

describe('PDFAbout.vue', () => {
  beforeEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })
  const component = PDFAbout
  // const props = Object.keys(component.props)
  const propDicts = [
    {prop: 'name', selector: 'h2'},
    {prop: 'surname', selector: '#pdf-surname'},
    {prop: 'flags', selector: '#pdf-flags'},
    {prop: 'email', selector: 'a'},
    {prop: 'address', selector: '#pdf-address'},
    {prop: 'visas', selector: '#pdf-visas'},
  ]
  let mockProp = true
  
  const props = propDicts.map(d => d.prop)
  
  let propsObject = propsMocker(props)

  propsObject.propsData['socials'] = [        
    {
      provider: 'github',
      url: 'github',
    },
    {
      provider: 'linkedin',
      url: 'linkedin',
    },
    {
      provider: 'instagram',
      url: 'instagram',
    },
    {
      provider: 'twitter',
      url: 'twitter',
    },
    {
      provider: 'website',
      url: '/',
    }
  ]

  console.log(propsObject)

  let wrapper = createWrapper(component, propsObject)

  propDicts.forEach(dict => {
    it(`renders interests ${dict.selector} that matches ${dict.prop} prop`, () => {
      expect(wrapper.find(`${dict.selector}`).text()).toMatch(propMocker(`${dict.prop}`).propsData[`${dict.prop}`])
    })  
  })

  it('email href matches email prop', () => {
    // interesting that although snapshot includes deeply mounted child (pdfsocials)
    // the wrapper doesn't find those elements
    const prop = 'email'
    const propDict = propDicts.filter(d => d.prop === prop)[0]
    expect(wrapper.find(`${propDict.selector}`).attributes().href).toMatch(propMocker(`${prop}`).propsData[`${prop}`])
  })

  it('renders photo', () => {

    expect(wrapper.find("img").exists()).toBe(true)

  })

  it('matches snapshot', () => {

    let _propsObject = cloneDeep(propsObject)
    _propsObject.propsData.renderPDF = false

    wrapper = createWrapper(component, _propsObject)

    expect(wrapper.html()).toMatchSnapshot()
    
  })

  it('matches renderPDF snapshot', () => {

    let _propsObject = cloneDeep(propsObject)
    _propsObject.propsData.renderPDF = true

    wrapper = createWrapper(component, _propsObject)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
