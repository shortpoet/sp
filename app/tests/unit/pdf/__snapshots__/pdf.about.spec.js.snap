// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFAbout.vue matches renderPDF snapshot 1`] = `
<section id="about" class="resume-section">
  <div id="basic-info" class="d-flex flex-row justify-content-between align-items-center mt-1 ml-3">
    <div id="photo" class="d-flex flex-column align-items-center"><img src="" alt="profile pic" class="img-fluid img-profile rounded m-1 mb-2" style="height: 8rem; width: 7rem;"></div>
    <div id="name-contact" class="d-flex flex-column mt-1 mt-md-2 ml-2">
      <ul class="list-inline">
        <li>
          <h2 class="heading heading-2 mb-0" style="font-family: 'Saira Extra Condensed';">
            sample name
            <span id="pdf-surname" class="text-primary" style="font-family: 'Saira Extra Condensed';">sample surname</span></h2>
        </li>
        <li><span class="not-name"><a href="mailto:sample email" style="font-family: 'Open Sans';">
sample email
</a></span></li>
        <li><span id="pdf-address" class="not-name" style="font-family: 'Saira Extra Condensed';">
sample address
</span></li>
        <li><span id="pdf-flags" class="not-name">sample flags</span></li>
        <li><span id="pdf-visas" class="not-name" style="font-family: 'Saira Extra Condensed';">Valid Work Visas: sample visas</span></li>
      </ul>
    </div>
    <div id="socials" class="d-flex flex-column ml-auto mr-1 mr-md-3">
      <section id="pdf-socials">
        <ul class="list-inline list-social-icons mb-0 mt-0 mr-lg-2">
          <li class="social-item"><a href="github"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .65rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-github fa-stack-1x fa-inverse"></i></span> <span class="social-url d-inline" style="font-family: 'Open Sans'; font-size: .55rem;">github</span></a></li>
          <li class="social-item"><a href="linkedin"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .65rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-linkedin fa-stack-1x fa-inverse"></i></span> <span class="social-url d-inline" style="font-family: 'Open Sans'; font-size: .55rem;">linkedin</span></a></li>
          <li class="social-item"><a href="instagram"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .65rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-instagram fa-stack-1x fa-inverse"></i></span> <span class="social-url d-inline" style="font-family: 'Open Sans'; font-size: .55rem;">instagram</span></a></li>
          <li class="social-item"><a href="twitter"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .65rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-twitter fa-stack-1x fa-inverse"></i></span> <span class="social-url d-inline" style="font-family: 'Open Sans'; font-size: .55rem;">twitter</span></a></li>
          <li class="social-item"><a href="/"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .65rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-home fa-stack-1x fa-inverse"></i></span> <span class="social-url d-inline" style="font-family: 'Open Sans'; font-size: .55rem;">https://shortpoet.com</span></a></li>
        </ul>
      </section>
    </div>
  </div>
</section>
`;

exports[`PDFAbout.vue matches snapshot 1`] = `
<section id="about" class="resume-section">
  <div id="basic-info" class="d-flex flex-row justify-content-between align-items-center mt-1 ml-3">
    <div id="photo" class="d-flex flex-column align-items-center"><img src="" alt="profile pic" class="img-fluid img-profile rounded m-1 mb-2" style="height: 8rem; width: 7rem;"></div>
    <div id="name-contact" class="d-flex flex-column mt-1 mt-md-2 ml-2">
      <ul class="list-inline">
        <li>
          <h2 class="heading heading-2 mb-0" style="font-family: 'Saira Extra Condensed';">
            sample name
            <span id="pdf-surname" class="text-primary" style="font-family: 'Saira Extra Condensed';">sample surname</span></h2>
        </li>
        <li><span class="not-name"><a href="mailto:sample email" style="font-family: 'Open Sans';">
sample email
</a></span></li>
        <li><span id="pdf-address" class="not-name" style="font-family: 'Saira Extra Condensed';">
sample address
</span></li>
        <li><span id="pdf-flags" class="not-name">sample flags</span></li>
        <li><span id="pdf-visas" class="not-name" style="font-family: 'Saira Extra Condensed';">Valid Work Visas: sample visas</span></li>
      </ul>
    </div>
    <div id="socials" class="d-flex flex-column ml-auto mr-1 mr-md-3">
      <section id="pdf-socials">
        <ul class="list-inline list-social-icons mb-0 mt-0 mr-lg-2">
          <li class="social-item"><a href="github"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .85rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-github fa-stack-1x fa-inverse"></i></span> <span class="social-url d-none d-md-inline" style="font-family: 'Open Sans'; font-size: .55rem;">github</span></a></li>
          <li class="social-item"><a href="linkedin"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .85rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-linkedin fa-stack-1x fa-inverse"></i></span> <span class="social-url d-none d-md-inline" style="font-family: 'Open Sans'; font-size: .55rem;">linkedin</span></a></li>
          <li class="social-item"><a href="instagram"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .85rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-instagram fa-stack-1x fa-inverse"></i></span> <span class="social-url d-none d-md-inline" style="font-family: 'Open Sans'; font-size: .55rem;">instagram</span></a></li>
          <li class="social-item"><a href="twitter"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .85rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-twitter fa-stack-1x fa-inverse"></i></span> <span class="social-url d-none d-md-inline" style="font-family: 'Open Sans'; font-size: .55rem;">twitter</span></a></li>
          <li class="social-item"><a href="/"><span class="social-icon-layer fa-stack fa-lg" style="font-family: FontAwesome; font-size: .85rem;"><i class="fa fa-circle fa-stack-2x"></i> <i class="social-icon fa fa-home fa-stack-1x fa-inverse"></i></span> <span class="social-url d-none d-md-inline" style="font-family: 'Open Sans'; font-size: .55rem;">https://shortpoet.com</span></a></li>
        </ul>
      </section>
    </div>
  </div>
</section>
`;
